### ISSUE TITLE
Inconsistent version numbers

### ISSUE THREAD

**Issue by mcepl** [dask/distributed#5912]
Sorry, this is just a nitpick, but while packaging for openSUSE I have to make a lot of acrobatics around your inconsistency in numbering:

* tag is 2022.02.1, which is correctly reflected in the tarball and the name of the directory inside of the tarball
* but the internal version is 2022.2.1 (notice missing zero) in egg-info and the name of the egg-info directory (`distributed-2022.2.1-py3.9.egg-info`)

Could you please unify it?

**Comment by ian-r-rose** [comment]
Thanks for bringing this up @mcepl, I've also been hit by this difference. As I understand it, the stripping of leading zeros is as described in [PEP 440](https://peps.python.org/pep-0440/#integer-normalization). So, from my perspective, the best way forward is to start publishing the git tags without the leading zeros for the month.

What do you think @jsignell? Should we add a note about this to the release procedure?

**Comment by fjetter** [comment]
I think the zero padding was an aware decision https://github.com/dask/community/issues/100

**Comment by ian-r-rose** [comment]
Ah, right, I had forgotten about that discussion. I think every six months or so I see some issue where somebody has problem with the zero-padded vs non-zero-padded versions of the numbers, though I don't have receipts here.

**Comment by jsignell** [comment]
Right. It's been discussed, but that doesn't mean that we arrived at the right conclusion. If someone wants to open the issue of zero-padding back up we can definitely discuss it. I personally don't have much of an opinion.
